addons/mail/static/src/discuss/core/common/message_model_patch.js

```diff
@@ -69,5 +69,8 @@ const messagePatch = {
             mentionedRoles,
         });
     },
+    showSeenIndicator(thread) {
+        return this.isSelfAuthored && thread?.hasSeenFeature;
+    },
```

Comment:
Would document that `thread` is the thread being viewed, for clarity on what it means and we understand why we need to pass it rather than `message.thread`.